pkgs/test/config.nix

```diff
@@ -0,0 +1,21 @@
+{ lib, ... }:
+lib.recurseIntoAttrs {
+
+  # https://github.com/NixOS/nixpkgs/issues/175196
+  allowPkgsInPermittedInsecurePackages =
+    let pkgs = import ../.. {
+          config = {
+            permittedInsecurePackages =
+              tempAllow pkgs.authy "2.1.0" [ "electron-9.4.4" ];
+          };
+        };
+        # Allow with forgetting
+        tempAllow = p: v: pa:
+          lib.optionals (lib.assertMsg (p.version == v) "${p.name} is no longer at version ${v}, consider removing the tempAllow") pa;
```

Comment:
@roberth I don't know exactly how this test is evalulated, but will this assert actually trip (and fail the test) when it doesn't hold? I have prepared a PR #175206 to bump authy to 2.2.0 (and remove the need for electron_9), and I wouldn't want this to immediately break when it gets merged.

Reply:
All it needs to do is be strict in `p.version` and `v`, so the function below would suffice for the purpose of the test. I did forget to actually change it to that function, so that was stupid. Let me fix that.

Reply:
Future proofing in https://github.com/NixOS/nixpkgs/pull/175274/files

Reply:
> but will this assert actually trip (and fail the test) when it doesn't hold?

It's not a test for `tempAllow`. All we need to do here is to evaluate the arguments, to simulate the real `tempAllow` behavior. This creates the infinite recursion when reverting `lazyAttrsOf` -> `attrsOf`, and I've double checked just now.